src/Education/XamlGenerator/RuleHelpXamlTranslator.cs

```diff
@@ -0,0 +1,459 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Xml;
+
+namespace SonarLint.VisualStudio.Education.XamlGenerator
+{
+    /// <summary>
+    /// Translator for html content in rule definition to xaml
+    /// </summary>
+    /// <remarks>
+    /// Our assumption is generated xaml will be partial and will be under a xaml root by caller
+    /// This root will either be <FlowDocument> or <Section>
```

Comment:
Does it actually have to make any assumptions about how it is going to be used?

It would be better to document that it actually returns e.g. `the generated XAML will have a single XAML root element that is a <Block> / list of <Block>` (whichever it is).